test-lib-openshift.sh

```diff
@@ -169,6 +169,21 @@ function ct_os_get_pod_ip() {
   oc get pod "$pod_name" --no-headers -o custom-columns=IP:status.podIP
 }
 
+# ct_os_get_sti_build_logs
+# -----------------
+# Return logs from sti_build
+# Arguments: pod_name
+function ct_os_get_sti_build_logs() {
+  local pod_name="${1}"
+  # Print logs but do not failed. Just for traces
+  if oc get "bc/$pod_name"; then
+    oc logs "bc/$pod_name" || return 0
```

Comment:
Will this work? In all uses of the function `pod_prefix` is passed as the `pod_name` argument. Looking at how the other functions use `pod_prefix` (grep is involved) it does not seem like `oc` would be able to work with just the prefix.

Reply:
Well, I have tested it on s2i-python-container.
pod_name here is something like `pod/python-testing-1-596hh` but buildconfig, shortly `bc` is python-testing.

See log from OpenShift:
```bash
Waiting for python-testing pod becoming ready ....................................................................................... FAIL
NAME                          READY   STATUS      RESTARTS   AGE
pod/python-testing-1-596hh    0/1     Running     3          5m6s
pod/python-testing-1-build    0/1     Completed   0          6m24s
pod/python-testing-1-deploy   1/1     Running     0          5m45s

[..snip..]

NAME                                                REVISION   DESIRED   CURRENT   TRIGGERED BY
deploymentconfig.apps.openshift.io/postgresql       0          1         0         config,image(postgresql:10)
deploymentconfig.apps.openshift.io/python-testing   1          1         1         config,image(python-testing:latest)

NAME                                            TYPE     FROM         LATEST
buildconfig.build.openshift.io/python-testing   Source   Git@master   1

NAME                                        TYPE     FROM          STATUS     STARTED         DURATION
build.build.openshift.io/python-testing-1   Source   Git@7cbc596   Complete   5 minutes ago   1m25s

[..snip..]

svc/postgresql - 172.30.227.114:5432
  dc/postgresql deploys istag/postgresql:10
    deployment #1 waiting on image or update

http://python-testing-sclorg-test-7600.apps.core-serv-ocp.hosted.psi.rdu2.redhat.com (svc/python-testing)
  dc/python-testing deploys istag/python-testing:latest <-
    bc/python-testing source builds https://github.com/sclorg/django-ex.git#master on istag/python:2.7
```

But we can detect buildconfig in a function like here: https://github.com/sclorg/container-common-scripts/blob/master/test-lib-openshift.sh#L136

`oc get bc -o "custom-columns=NAME:.metadata.name,Ready:status.phase" | grep "${pod_prefix}" | sort -u | tail -n 1`